include/constants/battle_config.h

```diff
@@ -178,6 +178,7 @@
 #define B_POWDER_GRASS              GEN_7 // In Gen6+, Grass-type Pokémon are immune to powder and spore moves.
 #define B_STEEL_RESISTANCES         GEN_7 // In Gen6+, Steel-type Pokémon are no longer resistant to Dark-type and Ghost-type moves.
 #define B_THUNDERSTORM_TERRAIN      TRUE  // If TRUE, overworld Thunderstorm generates Rain and Electric Terrain as in Gen 8.
+#define B_SEMI_INVULNERABLE_CATCH   GEN_4 // In Gen5+, you cannot throw a ball against a Pokemon that is in a semi-invulnerable state (dig/fly/etc)
```

Comment:
```suggestion
#define B_SEMI_INVULNERABLE_CATCH   GEN_7 // In Gen4+, you cannot throw a ball against a Pokemon that is in a semi-invulnerable state (dig/fly/etc)
```

Reply:
According to Bulbapedia, this change started in Gen 4.
https://bulbapedia.bulbagarden.net/wiki/Semi-invulnerable_turn